File: plugins/OrchidPosts/src/main/java/com/eden/orchid/posts/pages/PermalinkPage.java
package com.eden.orchid.posts.pages;

import com.eden.orchid.api.resources.resource.OrchidResource;
import com.eden.orchid.api.theme.pages.OrchidPage;


public abstract class PermalinkPage extends OrchidPage {

    public PermalinkPage(OrchidResource resource, String key) {
        super(resource, key);
    }

    public PermalinkPage(OrchidResource resource, String key, String title) {
        super(resource, key, title);
    }

    public PermalinkPage(OrchidResource resource, String key, String title, String path) {
        super(resource, key, title, path);
    }

    public abstract String getPermalink();

}

File: plugins/OrchidPosts/src/main/kotlin/com/eden/orchid/posts/pages/PermalinkPage.kt
package com.eden.orchid.posts.pages

import com.eden.orchid.api.options.annotations.Option
import com.eden.orchid.api.resources.resource.OrchidResource
import com.eden.orchid.api.theme.pages.OrchidPage


abstract class PermalinkPage : OrchidPage {

    constructor(resource: OrchidResource, key: String) : super(resource, key) {}

    constructor(resource: OrchidResource, key: String, title: String) : super(resource, key, title) {}

    constructor(resource: OrchidResource, key: String, title: String, path: String) : super(resource, key, title, path) {}


    @Option
    lateinit var permalink: String

}

